tslint.json

```diff
@@ -1,140 +1,24 @@
-            "check-type-operator",
-            "check-type-cast",
-            "check-branch",
-            "check-operator",
-            "check-separator",
-            "check-preblock",
-            "check-type",
-            "check-decl"],
-        "no-any": true,
-        "no-unused-expression": true,
-        "no-unnecessary-initializer": true,
-        "await-promise": [true, "Thenable"],
-        "completed-docs": false,
-        "no-unexternalized-strings": {"options": [{
-            "signatures": [
-                "localize",
-                "nls.localize"
-            ],
-            "keyIndex": 0,
-            "messageIndex": 1
-        }]},
-        "linebreak-style": false,
-        "quotemark": false,
-        "newline-per-chained-call": false,
-        "member-ordering": false,
-        "no-parameter-properties": false,
-        "no-inferrable-types": false,
-        "strict-type-predicates": true,
-        "increment-decrement": [true, "allow-post"],
-        "no-implicit-dependencies": false,
-        "ordered-imports": false,
-        "variable-name": false,
-        "prefer-readonly": false,
-        "comment-format": false,
-        "align": false,
-        "no-import-side-effect": false,
-        "switch-default": false,
-        "trailing-comma": false,
-        "switch-final-break": false,
-        "strict-string-expressions": false,
-        "newline-before-return": false,
-        "object-literal-sort-keys": false,
-        "no-suspicious-comment": false,
-        "export-name": false,
-        "no-magic-numbers": false,
-        "no-unsafe-any": false,
-        "no-unbound-method": false,
-        "detect-child-process": false,
-        "match-default-export-name": false,
-        "no-unnecessary-class": false,
-        "function-name": false,
-        "interface-name": false,
-        "cyclomatic-complexity": false,
-        "strict-comparisons": false,
-        "no-http-string": false,
-        "static-this": false,
-        "non-literal-fs-path": false,
-        "prefer-function-over-method": false,
-        "max-func-body-length": false,
-        "no-angle-bracket-type-assertion": false,
-        "no-for-in": false,
-        "arrow-return-shorthand": false,
-        "no-require-imports": false,
-        "no-string-literal": false,
-        "prefer-type-cast": false,
-        "no-non-null-assertion": false,
-        "ban-ts-ignore": false,
-        "no-namespace": false,
-        "no-unnecessary-local-variable": false,
-        "prefer-conditional-expression": false,
-        "no-reference": false,
-        "no-inner-html": false,
-        "no-single-line-block-comment": false
-    },
-    "rulesDirectory": ["node_modules/tslint-microsoft-contrib", "node_modules/tslint-no-unused-expression-chai/rules"],
-    "linterOptions": {
-        "exclude": [
-            "node_modules/**",
-            "out/**",
-            "src/test/**",
-            "src/explorer/**"
-        ]
-    }
+{
+	"rules": {
```

Comment:
why are there so many changes to this file? seems odd to completely change all the coding styles in one PR

Reply:
Most of the preexisting lint rules were added relatively recently (last month). Before that it was a yet a different set of rules.

I'm working to add as many as I can back, but some of the rules require non-trivial changes to satisfy. We can decide if we want to hold on delivering functionality for these linting changes.

Reply:
I'll be honest Jeff, this is such a dramatic change from any coding style that we have seen that it makes this code a bit difficult to read\understand. A simple example is the lack of semicolons and required bracing for if statements.

Even the original code was more formatted than this syntax.

One of the key reasons that the TSLint configuration was turned up so high was to enforce "consistency" in both style and code architecture in the code with other TypeScript code that Microsoft owns.

VSCode, for example, actually has a git extension installed that requires your code to be formatted a certain way before you can push it even to your own fork\branch :)

We should agree on some compromise here.

---
In reply to: [430750778](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r430750778) [](ancestors = 430750778)

Reply:
Merging threads. Moving Sherry's comment here: "why are there so many changes to this file? seems odd to completely change all the coding styles in one PR".

---
In reply to: [431194797](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r431194797) [](ancestors = 431194797,430750778)